src/graph/entities/graph_entity.c

```diff
@@ -35,7 +38,7 @@ SIValue *GraphEntity_GetProperty
 
 	// e->attributes is NULL when dealing with an "intermediate" entity,
 	// one which didn't had its attribute-set allocated within the graph datablock.
-	if(e->attributes == NULL) {
+	if(e->attributes == &NULL_ATTRIBUTE_SET) {
  		// note that this exception may cause memory to be leaked in the caller
  		ErrorCtx_SetError("Attempted to access undefined attribute");
  		return ATTRIBUTE_NOTFOUND;
```

Comment:
I think this entire block can be removed. leaving us with just:
```c
return AttributeSet_Get(*e->attributes, attr_id);
```
Now that we have `attributes = NULL_ATTRIBUTE_SET` the call to `AttributeSet_Get(*e->attributes, attr_id);` will receive `NULL` which it knows how to handle.

Reply:
Done.